### PR TITLE
feat: remove roles for organisation

### PR DESCRIPTION
**Pull request by trini7y**
## Description:
This PR introduces a new endpoint to retrieve all roles in an organization.

## How should this be manually tested?
Send  `DELETE` request to `/api/v1/roles/{org_id}/role/{role-id}`

## Checklist of what you did:

- [x] Set up an endpoint for removing a single role
- [x] Implement OrganizationGuard
- [x] Authentication and Authorization of `roles/org_id}/role/{role-id} DELETE` endpoint 
- [x] Validate role ID format
- [x] Implement database query to check if the role is assigned to any organization
- [x] Implement soft delete operation for roles
- [x] Implement error handling for various scenarios (invalid ID, non-existent role, non-existent organization, role in use) 
- [x] Write unit tests for the role deletion endpoint
- [x] Test error scenarios (invalid ID, non-existent role, non-existent organization, role in use)
- [x] Ensure that only users with appropriate permissions can delete roles

## Reference Issues
[[FEAT] API Endpoint to Fetch All Roles within an Organisation](okekedesmond71/bac2-17-remove-role-within-an-organization-256)

### CODE REVIEW

**Comment by CFCIfe** [comment]
Correct your PR title to match the norm

`feat: remove ...`

File: src/modules/organisation-role/organisation-role.controller.ts [hunk 1]
```diff
@@ -89,6 +94,18 @@ export class OrganisationRoleController {
     }
   }
 
+  @Delete(':id/roles:roleId')
+  @UseGuards(AuthGuard, OwnershipGuard)
```
Comment (CFCIfe) [inline src/modules/organisation-role/organisation-role.controller.ts:98]: AuthGuards are implemented by default
```suggestion
@UseGuards(OwnershipGuard)
```

File: src/modules/organisation-role/organisation-role.controller.ts [hunk 2]
```diff
@@ -89,6 +94,18 @@ export class OrganisationRoleController {
     }
   }
 
+  @Delete(':id/roles:roleId')
```
Comment (CFCIfe) [inline src/modules/organisation-role/organisation-role.controller.ts:97]: This path looks incorrect.

```suggestion
@Delete(':id/roles/:roleId')
```

File: src/modules/organisation-role/organisation-role.controller.ts [hunk 3]
```diff
@@ -89,6 +94,18 @@ export class OrganisationRoleController {
...
+  @UseGuards(AuthGuard, OwnershipGuard)
+  @HttpCode(HttpStatus.OK)
+  @ApiOperation({ summary: 'Delete a role in an organisation' })
+  @ApiParam({ name: 'organisationId', required: true, description: 'ID of the organisation' })
```
Comment (CFCIfe) [inline src/modules/organisation-role/organisation-role.controller.ts:101]: You have 2 query params but only defined one.

File: src/modules/organisation-role/organisation-role.controller.ts [hunk 4]
```diff
@@ -89,6 +94,18 @@ export class OrganisationRoleController {
...
+  @ApiResponse({ status: 200, description: 'Role successfully removed' })
+  @ApiResponse({ status: 400, description: 'Invalid role ID format' })
+  @ApiResponse({ status: 404, description: 'Role not found' })
+  async remove(@Param('id') organisationId: string, @Param('roleId') roleId: string) {
```
Comment (CFCIfe) [inline src/modules/organisation-role/organisation-role.controller.ts:105]: Use a more descriptive controller name

```suggestion
  async removeRole(@Param('id') organisationId: string, @Param('roleId') roleId: string) {
```

File: src/modules/organisation-role/organisation-role.service.ts [hunk 5]
```diff
@@ -126,6 +132,55 @@ export class OrganisationRoleService {
...
+          role: { id: roleId },
+        },
+      },
+      relations: ['organisationUsers', 'organisationUsers.role'],
+    });
+
+    if (usersWithRole > 0) {
+      throw new BadRequestException('Role is currently assigned to users');
+    }
+    try {
```
Comment (CFCIfe) [inline src/modules/organisation-role/organisation-role.service.ts:155]: There is a response interceptor for this. You can remove everything within the catch block, and make your dB call outside the try block.

File: src/modules/organisation-role/organisation-role.controller.ts (hunk 2 above)
Comment (trini7y) [reply src/modules/organisation-role/organisation-role.controller.ts:97]: I will fix this.

**Pull request closed without merging**